docs/src/main/docs/guides/07_health_se_guide.adoc

```diff
@@ -0,0 +1,677 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Health Check SE Guide
+:description: Helidon health checks
+:keywords: helidon, health-checks, health, check
+
+This document describes basic examples that use both built-in and custom health checks with Helidon SE.
+
+== What you need
+
+[width=50%,role="flex, sm7"]
+|===
+|About 15 minutes
+|<<about/03_prerequisites.adoc,Helidon Prerequisites>>
+|===
+
+=== Create a sample SE project
+
+Generate the project sources using the Helidon SE Maven archetype.
+The result is a simple project that can be used for the examples in this guide.
+
+[source,bash,subs="attributes+"]
+.Run the Maven archetype
+----
+mvn archetype:generate -DinteractiveMode=false \
+    -DarchetypeGroupId=io.helidon.archetypes \
+    -DarchetypeArtifactId=helidon-quickstart-se \
+    -DarchetypeVersion={helidon-version} \
+    -DgroupId=io.helidon.examples \
+    -DartifactId=helidon-quickstart-se \
+    -Dpackage=io.helidon.examples.quickstart.se
+----
+
+=== Using the built-in health-checks
+
+Helidon has a set of built-in health checks can be optionally enabled to report various
+ health check statuses that are commonly used:
+
+* deadlock detection
+* available disk space
+* available heap memory
+
+This example will demonstrate how to use the built-in health checks.  These examples are all executed
+from the root directory of your project (helidon-quickstart-se)
+
+[source,xml]
+.Notice that the built-in health check dependency is already in the project's pom.xml file
+----
+<dependency>
+    <groupId>io.helidon.health</groupId>
+    <artifactId>helidon-health-checks</artifactId>
+</dependency>
+----
+
+[source,java]
+.Modify Main.java, import HealthCheckReponse, and replace the createRouting method with the following:
+----
+
+import org.eclipse.microprofile.health.HealthCheckResponse;  // <1>
+
+...
+
+private static Routing createRouting(Config config) {
+
+    HealthSupport health = HealthSupport.builder()
+      .addLiveness(HealthChecks.healthChecks())  // <2>
```

Comment:
We should fix this in the quickstart example. I think it still uses the deprecated `add` method. You can do that in this PR or a separate one.

Reply:
I can do in a separate PR unless Joe does it, I think he said he was going to rework the Quickstarts